testCases/methane-emissions/year.json

```diff
@@ -2,28 +2,28 @@
   "variables": {
     "emissions": [
       {
-        "type":"Agricultural Waste Burning",
-        "result": "let maxEntry = maximumBy (fun x -> x.emissions) (filter (fun x -> x.type == \"$emissions.type$\") tableData) in numToStr maxEntry.year",
+        "type": "Agricultural Waste Burning",
+        "result": "numToStr (foldr1 (fun x y -> if x.emissions > y.emissions then x else y) (filter (fun x -> x.type == \"$emissions.type$\") tableData)).year",
         "record": "highest"
       },
       {
-        "type":"Agriculture",
-        "result": "let maxEntry = maximumBy (fun x -> x.emissions) (filter (fun x -> x.type == \"$emissions.type$\") tableData) in numToStr maxEntry.year",
+        "type": "Agriculture",
+        "result": "numToStr (foldr1 (fun x y -> if x.emissions > y.emissions then x else y) (filter (fun x -> x.type == \"$emissions.type$\") tableData)).year",
         "record": "highest"
       },
       {
-        "type":"The Energy Sector",
-        "result": "let maxEntry = maximumBy (fun x -> x.emissions) (filter (fun x -> x.type == \"$emissions.type$\") tableData) in numToStr maxEntry.year",
+        "type": "The Energy Sector",
+        "result": "numToStr (foldr1 (fun x y -> if x.emissions > y.emissions then x else y) (filter (fun x -> x.type == \"$emissions.type$\") tableData)).year",
         "record": "highest"
       },
       {
-        "type":"Residential Commercial Other",
-        "result": "let maxEntry = maximumBy (fun x -> x.emissions) (filter (fun x -> x.type == \"$emissions.type$\") tableData) in numToStr maxEntry.year",
+        "type": "Residential Commercial Other",
+        "result": "numToStr (foldr1 (fun x y -> if x.emissions > y.emissions then x else y) (filter (fun x -> x.type == \"$emissions.type$\") tableData)).year",
         "record": "lowest"
       },
       {
-        "type":"Grassland Burning",
-        "result": "let maxEntry = maximumBy (fun x -> x.emissions) (filter (fun x -> x.type == \"$emissions.type$\") tableData) in numToStr maxEntry.year",
+        "type": "Grassland Burning",
+        "result": "numToStr (foldr1 (fun x y -> if x.emissions > y.emissions then x else y) (filter (fun x -> x.type == \"$emissions.type$\") tableData)).year",
```

Comment:
I was too hasty with these examples, sorry -- `maximumBy` is fine, but let's put it `util` so it doesn't have be included in the paper (and use `head` to implement the extraction of the first element).